## DATA IMPORT


In [79]:
import pandas as pd
import numpy as np

## Column Headings

In [80]:
# Print the columns of df
maternal.columns

Index(['Data As Of', 'Jurisdiction', 'Group', 'Subgroup', 'Year of Death',
       'Month of Death', 'Time Period', 'Month Ending Date', 'Maternal Deaths',
       'Live Births', 'Maternal Mortality Rate', 'Footnote'],
      dtype='object')

In [81]:
# Print the type of df
print(maternal.dtypes)

Data As Of                  object
Jurisdiction                object
Group                       object
Subgroup                    object
Year of Death                int64
Month of Death               int64
Time Period                 object
Month Ending Date           object
Maternal Deaths             object
Live Births                 object
Maternal Mortality Rate    float64
Footnote                    object
dtype: object


In [99]:
# Use the head method to print the first 5 rows of df
print(maternal.head())

  Data As Of   Jurisdiction  Group Subgroup  Year of Death  Month of Death  \
0    10/6/24  United States  Total    Total           2019               1   
1    10/6/24  United States  Total    Total           2019               2   
2    10/6/24  United States  Total    Total           2019               3   
3    10/6/24  United States  Total    Total           2019               4   
4    10/6/24  United States  Total    Total           2019               5   

       Time Period Month Ending Date Maternal Deaths Live Births  \
0  12 month-ending           1/31/19             660   3,787,776   
1  12 month-ending           2/28/19             653   3,783,489   
2  12 month-ending           3/31/19             657   3,771,682   
3  12 month-ending           4/30/19             668   3,772,235   
4  12 month-ending           5/31/19             706   3,767,999   

   Maternal Mortality Rate Footnote  
0                     17.4      NaN  
1                     17.3      NaN  
2       

### Data As Of column

In [100]:

maternal_death_column = maternal['Maternal Deaths']

# Print the maternal_death_column to verify
print(maternal_death_column)

0      660
1      653
2      657
3      668
4      706
      ... 
655    259
656    263
657    264
658    271
659    272
Name: Maternal Deaths, Length: 660, dtype: object


In [101]:
# Replace 'your_file.csv' with the actual path to your CSV file
df = pd.read_csv('maternal data #1.csv')

# Ensure the column name is correct
maternal_death_column_name = 'Maternal Deaths'  # Updated column name

# Check if the column exists in the DataFrame
if maternal_death_column_name in df.columns:
    # Assign the column to a variable
    maternal_death_column = df[maternal_death_column_name]
    
    # Convert the column to a numpy array, ensuring numeric values and handling missing values
    maternal_death = pd.to_numeric(maternal_death_column, errors='coerce').dropna().values
    
    # Print the numpy array to verify
    print(maternal_death)
    
    

[660. 653. 657. 668. 706. 720. 721. 723. 729. 739. 752. 754. 740. 740.
 751. 767. 763. 774. 793. 803. 831. 843. 837. 861. 893. 899. 894. 899.
 911. 904. 908. 987. 942. 894. 870. 817. 766. 755. 762. 758. 732. 716.
 704. 688. 689. 676. 669. 673. 677. 677. 679. 678. 706. 708.  98. 103.
 101. 107. 113. 116. 117. 113. 117. 112. 112. 111. 113. 112. 112. 106.
 109. 110. 103. 106. 105. 113. 109. 114. 111. 110. 116. 120. 118. 123.
 124. 133. 142. 144. 154. 163. 168. 171. 165. 162. 158. 156. 160. 150.
 139. 131. 123. 113. 106.  98. 102.  99.  96.  92.  92.  88.  91.  90.
  94.  96. 101. 108. 112. 116. 124. 124. 461. 453. 464. 469. 500. 511.
 510. 516. 514. 528. 547. 544. 528. 531. 541. 558. 549. 554. 574. 581.
 602. 607. 592. 607. 635. 635. 632. 625. 633. 625. 628. 686. 763. 800.
 822. 854. 877. 886. 871. 856. 835. 842. 823. 750. 650. 616. 607. 578.
 544. 545. 550. 553. 537. 528. 520. 506. 510. 503. 492. 488. 477. 471.
 466. 461. 474. 472. 101.  97.  92.  92.  93.  93.  93.  93.  97.  98.
  92. 

### Mean of Maternal Death

In [102]:
# Calculate and print the mean
mean = np.mean(maternal_death)
print(mean)

262.63431542461007


### Maximum and Minimum Deaths

In [103]:
# Calculate and print the maximum
maximum = np.max(maternal_death)
print(maximum)

987.0


In [104]:
# Calculate and print the minimum
minimum = np.min(maternal_death)
print(minimum)

0.0


### Find the month that had the highest mortality rate

In [115]:
# Ensure the column names are correct
maternal_death_column_name = 'Maternal Deaths'
month_ending_date_column_name = 'Month Ending Date'

# Convert the 'Month Ending Date' column to datetime format
df[month_ending_date_column_name] = pd.to_datetime(df[month_ending_date_column_name], format='%m/%d/%y')

# Print the first few rows of the DataFrame to verify the data
print(df.head())

# Check if the columns exist in the DataFrame
if maternal_death_column_name in df.columns and month_ending_date_column_name in df.columns:
    # Ensure the 'Maternal Deaths' column is numeric
    df[maternal_death_column_name] = pd.to_numeric(df[maternal_death_column_name], errors='coerce')
    
    # Print unique values in the 'Maternal Deaths' column to verify
    print(df[maternal_death_column_name].unique())
    
    # Find the row where 'Maternal Deaths' is 987
    max_deaths_row = df[df[maternal_death_column_name] == 987]
    
    # Extract the corresponding 'Month Ending Date'
    if not max_deaths_row.empty:
        max_deaths_date = max_deaths_row[month_ending_date_column_name].values[0]
        formatted_date = pd.to_datetime(max_deaths_date).strftime('%B %d, %Y')
        print(f"The month ending date with the maximum maternal deaths of 987 is: {formatted_date}")
    else:
        print("No month ending date found with maternal deaths of 987.")
else:
    print(f"One or both columns '{maternal_death_column_name}' and '{month_ending_date_column_name}' do not exist in the DataFrame.")

  Data As Of   Jurisdiction  Group Subgroup  Year of Death  Month of Death  \
0    10/6/24  United States  Total    Total           2019               1   
1    10/6/24  United States  Total    Total           2019               2   
2    10/6/24  United States  Total    Total           2019               3   
3    10/6/24  United States  Total    Total           2019               4   
4    10/6/24  United States  Total    Total           2019               5   

       Time Period Month Ending Date Maternal Deaths Live Births  \
0  12 month-ending        2019-01-31             660   3,787,776   
1  12 month-ending        2019-02-28             653   3,783,489   
2  12 month-ending        2019-03-31             657   3,771,682   
3  12 month-ending        2019-04-30             668   3,772,235   
4  12 month-ending        2019-05-31             706   3,767,999   

   Maternal Mortality Rate Footnote  
0                     17.4      NaN  
1                     17.3      NaN  
2       

# Quiz Questions

### 1: How do you import the pandas library in Python?

#### code: import pandas as pd

### 2: How do you read a CSV file named data.csv into a pandas DataFrame?

#### import pandas as pd

#### #Read the CSV file into a DataFrame
df = pd.read_csv('data.csv')

### 3: How do you display the first 5 rows of a DataFrame named df?

#### # Display the first 5 rows of the DataFrame
print(df.head())

### 4: How do you find the number of rows and columns in a DataFrame df?

#### # Get the number of rows and columns
rows, columns = df.shape

print("Number of rows:", rows)
print("Number of columns:", columns)


### 5: How do you print the names of all the columns in a DataFrame df?

#### # Print the names of all the columns
print(df.columns)


### 6: How do you get the summary information of the DataFrame df including data types and non-null values?


#### # Get the summary information of the DataFrame
df.info()


### 7: How do you select only the rows in DataFrame df where the column A is greater than 5?

#### # Select rows where column A is greater than 5
filtered_df = df[df['A'] > 5]


### 8: How do you select the columns A and B from the DataFrame df?

#### # Select columns A and B
selected_columns = df[['A', 'B']]


### 9: How do you rename the column A to Alpha in DataFrame df?

#### # Rename column A to Alpha
df = df.rename(columns={'A': 'Alpha'})


### 10: How do you replace all occurrences of the value 999 in DataFrame df with nan?

#### import numpy as np

#### # Replace all occurrences of 999 with NaN
df.replace(999, np.nan, inplace=True)


### 11: How do you drop all rows in DataFrame df that contain missing values?

#### # Drop all rows with missing values
df_clean = df.dropna()


### 12: How do you fill all missing values in DataFrame df with 0?

#### # Fill all missing values with 0
df_filled = df.fillna(0)


### 13: How do you calculate the mean of column A in DataFrame df?

#### # Calculate the mean of column A
mean_value = df['A'].mean()

print(mean_value)


### 14: How do you create a new column C in DataFrame df that is the sum of columns A and B?

#### # Create a new column C as the sum of columns A and B
df['C'] = df['A'] + df['B']


### 15: How do you save the DataFrame df to a CSV file named output.csv? (Find the solution by searching the pandas documentation.)

#### import pandas as pd

#### #Assuming df is your DataFrame
df.to_csv('output.csv', index=False)
